### PR TITLE
xbutil program for edge

### PR DESCRIPTION
**Pull request by Himanshu-xilinx**

### CODE REVIEW

File: src/runtime_src/core/edge/tools/xbutil/xbutil.h [hunk 1]
```diff
@@ -524,7 +524,47 @@ class device {
         return 0;
     }
 
-    int program(const std::string& xclbin, unsigned region) { std::cout << "Unsupported API " << std::endl; return -1; }
+/*
+ *      * program
+ *           */
```
Comment (mamin506) [inline src/runtime_src/core/edge/tools/xbutil/xbutil.h]: /*
 * program
 */

File: src/runtime_src/core/edge/tools/xbutil/xbutil.h [hunk 2]
```diff
@@ -524,7 +524,47 @@ class device {
...
+            if (std::strncmp(temp, "xclbin2", 8))
+                return -EINVAL;
+        }
+
+
+        stream.seekg(0, stream.end);
+        int length = stream.tellg();
+        stream.seekg(0, stream.beg);
+
+        char *buffer = new char[length];
```
Comment (stsoe) [inline src/runtime_src/core/edge/tools/xbutil/xbutil.h]: Unmanaged new / delete are not a good idea.

  std::vector<char> header(length);
  stream.read(header.data(),length);

**Comment by abalasa** [comment]
retest this please

**Comment by abalasa** [comment]
retest this please

**Pull request merged**